### ISSUE TITLE
Include operational transactions first

### ISSUE THREAD

**Issue by pepyakin** [paritytech/substrate#3280]
Proposer can cut off transactions while building a block when reached deadline. To avoid excluding important transactions we should make sure that we put them into the block first.

**Comment by pepyakin** [comment]
Actually this is done in #3102 by setting the maximal priority for those.

**Comment by kianenigma** [comment]
Though I should mention that:

You (and/or @andresilva) recently brought up also the BABE time slots being potentially too short to fill a saturated block. Looks like we currently have three bounds to create a block and whichever gets exhausted first, will cause failure in including further transactions. 

#### Runtime checked. 

1. block size (5MB)
2. weight (custom quantity --for now the values are such that you can fit a few thousand instances of most transactions in it, I don't assume it will be the bottleneck).

#### Consensus checked.

3. Block proposer time limit: 1.6 seconds IIRC.

Ideally, in my opinion, the entire first group should have values that respect the consensus expected time to author a block. For now, we assumed that a node has 2 seconds to bake a block. We also took 1 second out of it as _emergency time_. Though', I did this because there some indications that out benchmarking is a bit too optimistic (e.g. indicating a few nanoseconds as the execution time of some transactions which I think is impossible). 

In other words, I think most of the time authoring code should be warned and stopped by the runtime checks that the block can no longer get bigger. The author then seals and finalizes baking the block. 

While the opposite, to my understanding, should be avoided. If the author runs out of time, it misses its slot and it will be completely wasted effort, right? 

If the above is true, I suppose the authoring code could actually stop the pipeline of validating transactions and adding them to the baking block at some point near the end (`t - 0.10 seconds`?).
